app/serializers/trip_serializer.rb

```diff
@@ -1,4 +1,14 @@
 class TripSerializer
   include JSONAPI::Serializer
   attributes :name, :location, :start_date, :end_date, :status, :total_budget
+
+  attribute :total_expenses do |object|
+    object.activities.sum(&:expenses) || 0
+  end
+
+  attribute :daily_itineraries do |object|
+    object.daily_itineraries.each_with_object({}) do |day, daily_itineraries|
+      daily_itineraries[day.date] = day.activities
+    end
```

Comment:
There is a way to add a conditional to only make it show on a Show action, however I couldn't seem to get it working on my end, here is doc code block from jsonapi-serializer:

```
class MovieSerializer
  include JSONAPI::Serializer

  attribute :director, if: Proc.new { |record, params|
    # The director will be serialized only if the :admin key of params is true
    params && params[:admin] == true
  }

  # Custom attribute `name_year` will only be serialized if both `name` and `year` fields are present
  attribute :name_year, if: Proc.new { |record|
    record.name.present? && record.year.present?
  } do |object|
    "#{object.name} - #{object.year}"
  end
end

# ...
current_user = User.find(cookies[:current_user_id])
serializer = MovieSerializer.new(movie, { params: { admin: current_user.admin? }})
serializer.serializable_hash
```

Reply:
This would be neat, if you can get this to work with just the show action, but if you need to pass it to all, I don't think it should be a big deal, since we can just filter the information we parse to the user in the FE